Fix/systemd unit tests

89luca89 opened the pull request:
Failing tests are because systemd-tests cannot detect us as a container-like solution and assumes we're a full system
So we manually skip what previously was automatically skipped by their checks.

Also we will add to the init the creation of `/.dockerenv` to trick other software to treat us more like a container than a full VM

Fix: https://github.com/chainguard-dev/prodsec/issues/313

zmarano commented on systemd.yaml:
```diff
@@ -181,7 +182,8 @@ subpackages:
             -s test-fstab-generator.sh \
-            -s test-rm-rf
+            -s test-rm-rf \
+            -s test-execute
```
Looks like you need to also skip `test-mount-util` and `test-namespace`

89luca89 replied:
I see they're failing because first they need the new microvm-init, I'll explicitly add them

89luca89 replied:
Now PR depends on https://github.com/wolfi-dev/os/pull/56962

89luca89 committed:
fix: skip test-execute in systemd-test

This test is skipped in all containerization solutions, our microvm is
not detected as such, so let's manually skip this

Signed-off-by: Luca Di Maio <luca.dimaio1@gmail.com>

89luca89 committed:
Merge branch 'main' into fix/systemd_unit_tests
